lib/spack/spack/test/environment_modifications.py

```diff
@@ -62,18 +63,28 @@ def test_exclude_paths_from_inspection():
     assert len(env) == 0
 
 
+def begin_esc():
```

Comment:
I think the usage of this method in this test module is not properly testing the expected behavior of the test module. On Linux/Mac the `/` character is the root of the filesystem in addition to the path separator, however on Windows those roles are the drive component and the backslash. The previous linux behavior uses the `/` because its testing absolute paths as members of path environment variables, so the Windows conversion should mirror that behavior.